### ISSUE TITLE
Long values fail to serialize into JSON

### ISSUE THREAD

**Issue by Albert221** [wk8/go-ordered-map#17]
```go
	t.Run("int key", func(t *testing.T) {
		om := New[int, any]()
		om.Set(1, "bar")
		om.Set(7, "baz")
		om.Set(2, 28)
		om.Set(3, 100)
		om.Set(4, "baz")
		om.Set(5, "28")
		om.Set(6, "100")
		om.Set(8, "baz")
		om.Set(8, "baz")
		om.Set(9, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque auctor augue accumsan mi maximus, quis viverra massa pretium. Phasellus imperdiet sapien a interdum sollicitudin. Duis at commodo lectus, a lacinia sem.")

		b, err := json.Marshal(om)
		assert.NoError(t, err)
		assert.Equal(t, `{"1":"bar","7":"baz","2":28,"3":100,"4":"baz","5":"28","6":"100","8":"baz", "9": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque auctor augue accumsan mi maximus, quis viverra massa pretium. Phasellus imperdiet sapien a interdum sollicitudin. Duis at commodo lectus, a lacinia sem."}`, string(b))
	})
```

```
--- FAIL: TestMarshalJSON (0.00s)
    --- FAIL: TestMarshalJSON/int_key (0.00s)
        json_test.go:52: 
                Error Trace:    json_test.go:52
                Error:          Received unexpected error:
                                json: error calling MarshalJSON for type *orderedmap.OrderedMap[int,interface {}]: invalid character 'g' in literal null (expecting 'u')
                Test:           TestMarshalJSON/int_key
        json_test.go:53: 
                Error Trace:    json_test.go:53
                Error:          Not equal: 
                                expected: "{\"1\":\"bar\",\"7\":\"baz\",\"2\":28,\"3\":100,\"4\":\"baz\",\"5\":\"28\",\"6\":\"100\",\"8\":\"baz\", \"9\": \"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque auctor augue accumsan mi maximus, quis viverra massa pretium. Phasellus imperdiet sapien a interdum sollicitudin. Duis at commodo lectus, a lacinia sem.\"}"
                                actual  : ""
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1 +1 @@
                                -{"1":"bar","7":"baz","2":28,"3":100,"4":"baz","5":"28","6":"100","8":"baz", "9": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque auctor augue accumsan mi maximus, quis viverra massa pretium. Phasellus imperdiet sapien a interdum sollicitudin. Duis at commodo lectus, a lacinia sem."}
                                +
                Test:           TestMarshalJSON/int_key
FAIL
```

I believe this is due to returning `jwriter.Writer.Buffer.Buf` in `MarshalJSON`

**Comment by wk8** [comment]
Ooops I'm sorry. Fixed in v2.1.1, https://github.com/wk8/go-ordered-map/commit/ea3e882d675798792bd3f9e39c5b491f3fb0d3c6

**Comment by Albert221** [comment]
That helps, thanks!
